tests/error/errors_on_usage/and_not_defined.err

```diff
@@ -1,7 +1,11 @@
-Guppy compilation failed. Error in file $FILE:9
+Error: Variable not defined (at $FILE:9:15)
+  | 
+7 |         return z
+8 |     else:
+9 |         return z
+  |                ^ `z` might be undefined ...
+  | 
+6 |     if x and (z := y + 1):
+  |        - ... if this expression is `False`
```

Comment:
to my eye this looks like `if x False` rather than `if x and (z := y + 1) is False` ...

Reply:
I think the message is correct:

If `x` were true then `z` would be defined because of the walrus assignment even if the whole thing tends out to be false.

If `x` is false, then `z` is not defined because `and` short circuits

Reply:
This is the test:
```python
@guppy
def foo(x: bool, y: int) -> int:
    if x and (z := y + 1):
        return z
    else:
        return z
```

Reply:
that makes sense! thanks